// @vercel/oidc/dist/exchange-vercel-oidc-token.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var exchange_vercel_oidc_token_exports = {};
__export(exchange_vercel_oidc_token_exports, {
  exchangeVercelOidcToken: () => exchangeVercelOidcToken
});
module.exports = __toCommonJS(exchange_vercel_oidc_token_exports);
var import_version = require("./version");
async function exchangeVercelOidcToken(options) {
  const response = await fetch("https://oidc.vercel.com/~token", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      Accept: "application/json",
      "User-Agent": `@vercel/oidc@${import_version.version}`
    },
    body: JSON.stringify({
      token: options.token,
      aud: options.audience,
      ...options.jti ? { jti: options.jti } : void 0
    })
  });
  if (!response.ok) {
    throw new Error(
      `Failed to exchange token: ${await readErrorMessage(response)}`
    );
  }
  let data;
  try {
    data = await response.json();
  } catch (_error) {
    throw new Error("Failed to exchange token: response was not valid JSON");
  }
  if (!data || typeof data !== "object" || !("token" in data) || typeof data.token !== "string") {
    throw new Error(
      "Failed to exchange token: response did not contain a token"
    );
  }
  return data.token;
}
async function readErrorMessage(response) {
  try {
    const data = await response.json();
    if (data && typeof data === "object" && "error" in data && typeof data.error === "string") {
      return data.error;
    }
  } catch (_error) {
  }
  return response.statusText || `HTTP ${response.status}`;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  exchangeVercelOidcToken
});
